api/graphql/schema.graphql

```diff
@@ -2275,7 +2276,6 @@ type Mutation {
     jobPosterTemplate: CreateJobPosterTemplateInput! @spread
   ): JobPosterTemplate @create @guard @canModel(ability: "create")
   updateJobPosterTemplate(
```

Comment:
Not having the ID the first parameter of the update mutation is a bit out of convention.  What is this change working around?  Can we solve it a different way?

Reply:
Because we're running validation on the UpdateJobPosterTemplateInput type the id doesn't get passed in to validation using the other method. This means I can't access it to view current data and even that the unique check for referenceId will stumble on itself if you pass in it's current value when updating.
You can also see it in the update user mutations where we actually pass in the id in both places and if you miss passing the id in the input type it can cause the email unique check to fail on itself.

Reply:
For example this mutation
```
mutation UpdateUser($id: ID!, $user: UpdateUserAsUserInput!) {
  updateUserAsUser(id: $id, user: $user) {
    id
    email
  }
}
```

will fail if you fail sending the id a second time here
```
{
  "id": "16000f9d-41f7-4912-bd3d-a9b92f3f77d6",
  "user": {
    "id": "16000f9d-41f7-4912-bd3d-a9b92f3f77d6",
    "email": "admin@test.com"
  }
}
```

Reply:
Ah, OK.  Fair enough.